### PR TITLE
Change class component to function and add hooks

### PR DESCRIPTION
**Pull request by Leandro-Barretoo**
In this PR I change the Calculator class to a functional component and add hooks instead of the class state.
Thank you for the review.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,62 +1,54 @@
...
+  const [total, setTotal] = useState(null);
+  const [next, setNext] = useState(null);
 
-  handler(e) {
-    this.setState((state) => calculate(state, e.target.innerText));
-  }
+  const changeState = (e) => {
+    setTotal(calculate(total, e.target.innerText).total);
+    setNext(calculate(next, e.target.innerText).next);
+  };
```
Comment (Dipeshtwis) [inline src/components/Calculator.js]: - [ ] I'm trying to click the first number and I'm getting this error that `obj is null`. Please check the same in the screenshot and try to fix it. I think there is a problem with the `next` state.

Screenshot from 2021-08-20 04-25-52

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @pasytchangwa, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: Refactored the Calculator component (components/Calculator.js) from a class-based component to a functional component with hooks.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

- [ ] I can see the project title, description is not matching to the current project. Please update it to make the README professional. Also, add the steps on how to run the project in local like 

```
git clone https://yourrepo.git
cd yourrepo
npm install
npm start
```
- [ ] Currently, the live link is pointing to another site that doesn't exist. please fix it.

**_Also check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Leandro-Barretoo** [comment]
I fixed the hook and updated the readme file with more descriptive information.

File: README.md [hunk 2]
```diff
@@ -1,57 +1,44 @@
...
-- Frameworks
-- Technologies used
+- HTML5, CSS3
+- JavaScript
+- React
 
 ## Live Demo
 
-[Live Demo Link](https://livedemo.com)
+[Live Demo Link](https://leandro-barretoo.github.io/math_magicians/build)
```
Comment (bellom) [inline README.md]: Thanks for updating the live link but the link added is pointing to page `404`

image

Kindly fix

File: src/components/Calculator.js [hunk 3]
```diff
@@ -3,19 +3,17 @@ import './Calculator.css';
 import calculate from '../logic/calculate';
 
 const Calculator = () => {
-  const [total, setTotal] = useState(null);
-  const [next, setNext] = useState(null);
+  const [state, myState] = useState({ total: null, next: null, operation: null });
 
   const changeState = (e) => {
```
Comment (bellom) [inline src/components/Calculator.js:8]: Could this be coming from my end 🤦, your app seems not to work properly, it takes 2-3 clicks for a successful click.
i.e I have to click the button 3 times.
And also I can't perform a successful operation like `5+5`, I got the following error 👇 

image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @Leandro-Barretoo  👏 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Leandro-Barretoo** [comment]
I deployed the app with gh-pages and updated the live link.
On my end, I don't have the invalid number error, so I cannot fix it.
Thank you for the review.

**Review by ZawadiSumba66 — APPROVED** [review]
Hi @Leandro-Barretoo 👋 , 
## STATUS: APPROVED 👍 
Great job implementing the changes requested by the previous reviewers.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### NOTE:
Kindly correct the live demo link as requested by the other reviewers.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Leandro-Barretoo** [comment]
Fix live link

**Pull request merged**